api/src/test/java/org/apache/iceberg/TestPartitionSpecValidation.java

```diff
@@ -36,95 +37,86 @@ public class TestPartitionSpecValidation {
 
   @Test
   public void testMultipleTimestampPartitions() {
-    AssertHelpers.assertThrows(
-        "Should not allow year(ts) and year(ts)",
-        IllegalArgumentException.class,
-        "Cannot use partition name more than once",
-        () -> PartitionSpec.builderFor(SCHEMA).year("ts").year("ts").build());
-    AssertHelpers.assertThrows(
-        "Should not allow year(ts) and month(ts)",
-        IllegalArgumentException.class,
-        "Cannot add redundant partition",
-        () -> PartitionSpec.builderFor(SCHEMA).year("ts").month("ts").build());
-    AssertHelpers.assertThrows(
-        "Should not allow year(ts) and day(ts)",
-        IllegalArgumentException.class,
-        "Cannot add redundant partition",
-        () -> PartitionSpec.builderFor(SCHEMA).year("ts").day("ts").build());
-    AssertHelpers.assertThrows(
-        "Should not allow year(ts) and hour(ts)",
-        IllegalArgumentException.class,
-        "Cannot add redundant partition",
-        () -> PartitionSpec.builderFor(SCHEMA).year("ts").hour("ts").build());
-
-    AssertHelpers.assertThrows(
-        "Should not allow month(ts) and month(ts)",
-        IllegalArgumentException.class,
-        "Cannot use partition name more than once",
-        () -> PartitionSpec.builderFor(SCHEMA).month("ts").month("ts").build());
-    AssertHelpers.assertThrows(
-        "Should not allow month(ts) and day(ts)",
-        IllegalArgumentException.class,
-        "Cannot add redundant partition",
-        () -> PartitionSpec.builderFor(SCHEMA).month("ts").day("ts").build());
-    AssertHelpers.assertThrows(
-        "Should not allow month(ts) and hour(ts)",
-        IllegalArgumentException.class,
-        "Cannot add redundant partition",
-        () -> PartitionSpec.builderFor(SCHEMA).month("ts").hour("ts").build());
-
-    AssertHelpers.assertThrows(
-        "Should not allow day(ts) and day(ts)",
-        IllegalArgumentException.class,
-        "Cannot use partition name more than once",
-        () -> PartitionSpec.builderFor(SCHEMA).day("ts").day("ts").build());
-    AssertHelpers.assertThrows(
-        "Should not allow day(ts) and hour(ts)",
-        IllegalArgumentException.class,
-        "Cannot add redundant partition",
-        () -> PartitionSpec.builderFor(SCHEMA).day("ts").hour("ts").build());
-
-    AssertHelpers.assertThrows(
-        "Should not allow hour(ts) and hour(ts)",
-        IllegalArgumentException.class,
-        "Cannot use partition name more than once",
-        () -> PartitionSpec.builderFor(SCHEMA).hour("ts").hour("ts").build());
+    Assertions.assertThatThrownBy(
+            () -> PartitionSpec.builderFor(SCHEMA).year("ts").year("ts").build())
+        .hasMessageContaining("Cannot use partition name more than once")
+        .isInstanceOf(IllegalArgumentException.class);
```

Comment:
nit: to be consistent with other places, the `.hasMessage..` part comes after `.isInstanceOf()`